compiler/rustc_mir_build/src/builder/expr/into.rs

```diff
@@ -281,6 +283,33 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 this.diverge_from(block);
                 success.unit()
             }
+            ExprKind::ByUse { expr, span } => {
+                let place = unpack!(block = this.as_place(block, expr));
+                let ty = place.ty(&this.local_decls, this.tcx).ty;
+
+                let success = this.cfg.start_new_block();
```

Comment:
I know this is going to change in future commits, but it needs a comment. Just something like

```rust
// Convert `expr.use` to a call like `Clone::clone(&expr)`
```

would go a long way